lua/oil/config.lua

```diff
@@ -204,9 +206,14 @@ local default_config = {
 -- The adapter API hasn't really stabilized yet. We're not ready to advertise or encourage people to
 -- write their own adapters, and so there's no real reason to edit these config options. For that
 -- reason, I'm taking them out of the section above so they won't show up in the autogen docs.
+
+-- not "oil-s3://" on older neovim versions, since it doesn't open buffers correctly with a number
+-- in the name
+local oil_s3_string = vim.fn.has("nvim-0.12") and "oil-s3://" or "oil-sss://"
```

Comment:
`has()` returns a 1 for true and 0 for false, which are both truthy in lua. You'll want to do `vim.fn.has("nvim-0.12") == 1 and ...`